src/crimson/osd/osd.cc

```diff
@@ -433,17 +433,29 @@ seastar::future<> OSD::store_maps(ceph::os::Transaction& t,
       OSDMap::Incremental inc;
       auto i = p->second.cbegin();
       inc.decode(i);
-      return load_map_bl(e - 1)
-        .then([&t, e, inc=std::move(inc), this](bufferlist bl) {
-          auto o = std::make_unique<OSDMap>();
-          o->decode(bl);
-          o->apply_incremental(inc);
-          bufferlist fbl;
-          o->encode(fbl, inc.encode_features | CEPH_FEATURE_RESERVED);
-          store_map_bl(t, e, std::move(fbl));
-          osdmaps.insert(e, std::move(o));
-          return seastar::now();
-      });
+      if (e >1){
```

Comment:
under what circumstances we can have an incremental `1` epoch ?

Reply:
after --mkfs,  collection is empty. can't read any osdmap.  this is the existing ceph-osd logic, still has other logic such as when e >1, read fun return not found. but I don't know exactly when it is triggered. we ignore all of them in crimson-osd.

Reply:
after --mkfs, the incremental osdmap start from epoch 1.

Reply:
got it. but personally, i am not in favor of repeating. could you use `seastar::do_with()` instead?